src/zenml/orchestrators/utils.py

```diff
@@ -61,11 +105,27 @@ def execute_step(
     Returns:
         Optional execution info returned by the launcher.
     """
-    step_name = tfx_launcher._pipeline_node.node_info.id  # type: ignore[attr-defined] # noqa
+    step_name_param = (
+        INTERNAL_EXECUTION_PARAMETER_PREFIX + PARAM_PIPELINE_PARAMETER_NAME
+    )
+    pipeline_step_name = tfx_launcher._pipeline_node.node_info.id  # type: ignore[attr-defined]
     start_time = time.time()
-    logger.info(f"Step `{step_name}` has started.")
+    logger.info(f"Step `{pipeline_step_name}` has started.")
     try:
         execution_info = tfx_launcher.launch()
+        if execution_info and get_cache_status(execution_info):
+            pipeline_run_id = execution_info.pipeline_run_id
+            if execution_info.exec_properties:
+                step_name = json.loads(
+                    execution_info.exec_properties[step_name_param]
+                )
+            else:
+                logger.error(
+                    f"No execution properties found for step `{pipeline_step_name}`."
+                )
+            logger.info(
+                f"Using cached version of `{pipeline_step_name}` [`{step_name}`] from pipeline_run_id `{pipeline_run_id}`."
```

Comment:
Maybe I am misunderstanding the flow here, but wouldn't `step_name` not exist if it goes into the `else` condition above? Perhaps it might make sense to just but this log in the `if` , and with the `else` raise an exception?